core/src/main/java/io/opencensus/stats/Aggregate.java

```diff
@@ -0,0 +1,254 @@
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.Function;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * {@link Aggregate} is the result of applying a given {@link Aggregation} to a set of
+ * {@code MeasureValue}s.
+ *
+ * <p>{@link Aggregate} currently supports 6 types of basic aggregation:
+ * <ul>
+ *   <li>SUM
+ *   <li>COUNT
+ *   <li>HISTOGRAM
+ *   <li>RANGE
+ *   <li>MEAN
+ *   <li>STDDEV (standard deviation)
+ * </ul>
+ *
+ * <p>{@link ViewData} will contain one or more {@link Aggregate}s, corresponding to its
+ * {@link Aggregation} definition in {@link View}.
+ */
+@Immutable
+public abstract class Aggregate {
+
+  private Aggregate() {
+  }
+
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  public abstract <T> T match(
+      Function<? super AggregateSum, T> p0,
+      Function<? super AggregateCount, T> p1,
+      Function<? super AggregateHistogram, T> p2,
+      Function<? super AggregateRange, T> p3,
+      Function<? super AggregateMean, T> p4,
+      Function<? super AggregateStdDev, T> p5);
+
+  /** The sum value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateSum extends Aggregate {
+
+    AggregateSum() {
+    }
+
+    static AggregateSum create(double sum) {
+      return new AutoValue_Aggregate_AggregateSum(sum);
+    }
+
+    /**
+     * Returns the aggregated sum.
+     *
+     * @return the aggregated sum.
+     */
+    public abstract double get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p0.apply(this);
+    }
+  }
+
+  /** The count value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateCount extends Aggregate {
+
+    AggregateCount() {
+    }
+
+    static AggregateCount create(long count) {
+      return new AutoValue_Aggregate_AggregateCount(count);
+    }
+
+    /**
+     * Returns the aggregated count.
+     *
+     * @return the aggregated count.
+     */
+    public abstract long get();
```

Comment:
I just thought it would be shorter and cleaner to just name these methods `get()`. For example, personally I'll perfer `Sum.get()` than `Sum.getSum()`. It makes sense for `Range` to have getters `getMin()` and `getMax` though.